src/commands/dao_cmds/call.js

```diff
@@ -0,0 +1,113 @@
+const { listApps } = require('./utils/knownApps')
+const { rolesForApps } = require('./acl_cmds/utils/knownRoles')
+const { ensureWeb3 } = require('../../helpers/web3-fallback')
+import { addressesEqual } from './utils/web3-utils'
+
+const Table = require('cli-table')
+const colors = require('colors')
+
+const knownRoles = rolesForApps()
+const ANY_ENTITY = '0xFFfFfFffFFfffFFfFFfFFFFFffFFFffffFfFFFfF'
+const ANY_ENTITY_TEXT = 'Any entity'
+
+const path = require('path')
+let knownApps
+
+exports.command = 'call <dao> <proxy-address> <fn> [fn-args]'
+
+exports.describe = 'Inspect permissions in a DAO'
+
+exports.builder = function (yargs) {
+  return daoArg(yargs)
+  .positional('proxy-address', {
+    description: 'Proxy address of the app with the function to be run'
+  })
+  .positional('fn', {
+    description: 'Function to be called'
+  })
+  .option('fn-args', {
+    description: 'Arguments to be passed to the function',
+    array: true,
+    default: [],
+  })
+}
+
+exports.handler = async function ({ reporter, dao, apm, network, proxyAddress, fn, fnArgs }) {
+  knownApps = listApps()
+  const web3 = await ensureWeb3(network)
+
+  const tasks = new TaskList([
+    {
+      title: 'Generating transaction',
+      task: async (ctx, task) => {
+        task.output = `Fetching dao at ${dao}...`
+
+
+        if (!ctx.accounts) {
+          ctx.accounts = await web3.eth.getAccounts()
+        }
+
+        return new Promise((resolve, reject) => {
+
+
+          initAragonJS(dao, apm['ens-registry'], {
+            accounts: ctx.accounts,
+            provider: web3.currentProvider,
+            onTransaction: transaction => {
+              ctx.transaction = transaction
+            },
+            onPermissions: permissions => {
+              ctx.acl = permissions
+            },
+            onApps: async apps => {
+              ctx.apps = apps
+              if (ctx.wrapper) {
+                ctx.transactionPath = await ctx.wrapper.getTransactionPath(proxyAddress, fn, fnArgs)
+                resolve()
+              }
+            },
+            onDaoAddress: addr => ctx.daoAddress = addr,
+            onError: err => reject(err)
+          })
+            .then(async wrapper => {
+              ctx.wrapper = wrapper
+              if (ctx.apps) {
+                ctx.transactionPath = await ctx.wrapper.getTransactionPath(proxyAddress, fn, fnArgs)
+                resolve()
+              }
+            })
+            .catch(err => {
+              reporter.error('Error inspecting DAO')
+              reporter.debug(err)
+              process.exit(1)
+            })
+        })
+      }
+    },
+    {
+      title: `calling ${fn}`,
+      task: async (ctx, task) => {
+        task.output = `Waiting for response...`
+        ctx.transactionPath[0].gas = await web3.eth.estimateGas(ctx.transactionPath[0])
+        ctx.transactionPath[0].gas = ctx.transactionPath[0].gas *2 // need better estimations
```

Comment:
Would rewrite to:

```js
let tx = ctx.transactionPath[0] // TODO: Support choosing between possible transaction paths

if (!tx) {
  throw new Error('Cannot find transaction path for executing action')
}

const estimatedGas = await web3.eth.estimateGas(ctx.transactionPath[0])
tx.gas = parseInt(GAS_ESTIMATE_ERROR_FACTOR * estimatedGas)
```

`const GAS_ESTIMATE_ERROR_FACTOR` should be defined near `ANY_ENTITY`